[vscode][proposed] Stub onWillCreateEditSessionIdentity

marcdumais-work opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md

Note: Security vulnerabilities should not be disclosed on GitHub, through a PR or any
other means. See SECURITY.md at the root of this repository, to learn how to report
vulnerabilities.
-->

#### What it does
<!-- Include relevant issues and describe how they are addressed. -->
This event was recently added to proposed API `EditSessionIdentityProvider`. For now we add a stub that will allow `vscode.git@1.77.0` and more recent to activate and be successfully used in Theia applications.

Fixes #12521

#### How to test
<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->
You can use the vscode.git 1.77.0  extensions below: 
[git-.zip](https://github.com/eclipse-theia/theia/files/11481204/git-.zip)

- Extract the git extensions and copy them in the "plugins" folder. 
- start the example application, open a workspace
- `vscode.git` should activate, not complaining about `onWillCreateEditSessionIdentity` not existing. Built-in `git` will not ultimately work, unless `@theia/git` is removed from the app - but that's no necessary for this PR)

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review) 

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

tsmaeder reviewed (commented):
Woudn't the simplest stub implementation just be an empty function function? No need for emitters and such.

vince-fugnitto reviewed (approved):
LGTM :+1: 
I confirmed that the builtin activates as expected, and the error is not thrown:

```
Activating extension 'Git (built-in)' failed: r.workspace.onWillCreateEditSessionIdentity is not a function
```

marcdumais-work commented:
> Woudn't the simplest stub implementation just be an empty function function? No need for emitters and such.

That's correct. I went a bit further than the simplest. I would not mind going with something simpler.

paul-marechal commented on packages/plugin-ext/src/plugin/plugin-context.ts:
```diff
@@ -721,6 +722,15 @@ export function createAPIFactory(
+             * to use this proposed API.
+             */
+            get onWillCreateEditSessionIdentity(): theia.Event<theia.EditSessionIdentityWillCreateEvent> {
+                return workspaceExt.onWillCreateEditSessionIdentityEvent;
```
```suggestion
                return () => Disposable.NULL;
```

marcdumais-work commented:
@tsmaeder as discussed I updated the 1st commit to use a separate `proposed.d.ts` file instead of adding to `theia-proposed.d.ts`. I took a fresh copy from vscode@1.77.0, only slightly modifying it to adapt to our namespace. 

Then I added commits on top to simplify the stub and tidy-up.

paul-marechal commented on packages/plugin-ext/src/plugin/plugin-context.ts:
```diff
@@ -721,7 +722,14 @@ export function createAPIFactory(
+             * This is a stub implementation, that should minimally satisfy vscode built-in extensions 
+             * that currently use this proposed API.
+             */
+            // onWillCreateEditSessionIdentity: () => Disposable.create(() => {} ),
```
Left over?

```suggestion
```

marcdumais-work commented:
I will resolve the conflicts locally after the other proposed API stubs are merged.

vince-fugnitto commented:
@marcdumais-work it should be fine to rebase now :+1:

vince-fugnitto commented on packages/plugin-ext/src/plugin/plugin-context.ts:
```diff
@@ -735,7 +736,13 @@ export function createAPIFactory(
+            },
+            /**
+             * @stubbed
+             * This is a stub implementation, that should minimally satisfy vscode built-in extensions 
```
There seems to be a trailing whitespace that is failing the `lint`:

```suggestion
             * This is a stub implementation, that should minimally satisfy vscode built-in extensions
```

vince-fugnitto reviewed (approved):
LGTM :+1:
